frontend/components/PearlWallet/PearlWallet.tsx

```diff
@@ -66,34 +59,27 @@ const PearlWalletContent = () => {
       case STEPS.ENTER_WITHDRAWAL_ADDRESS:
         return <EnterWithdrawalAddress onBack={handleBack} />;
       case STEPS.DEPOSIT:
-        return <Deposit onBack={handleBack} onContinue={handleNext} />;
-      case STEPS.SELECT_PAYMENT_METHOD:
-        return <SelectPaymentMethod onBack={handleBack} />;
+        return <PearlDeposit onBack={handleBack} />;
       default:
-        throw new Error('Invalid step');
+        throw new Error('Invalid page');
     }
-  }, [step, handleBack, handleNext, updateStep]);
+  }, [step, handleNext, handleBack, updateStep]);
 
   return content;
 };
 
 export const PearlWallet = () => {
-  const { walletStep } = usePearlWallet();
+  const { walletStep: step } = usePearlWallet();
 
   return (
-    <PearlWalletProvider>
-      <Flex
-        vertical
-        style={{
-          width:
-            walletStep === STEPS.SELECT_PAYMENT_METHOD
-              ? undefined
-              : MAIN_CONTENT_MAX_WIDTH,
-          margin: '0 auto',
-        }}
-      >
-        <PearlWalletContent />
-      </Flex>
-    </PearlWalletProvider>
+    <Flex
+      vertical
+      style={{
+        width: step === STEPS.DEPOSIT ? undefined : MAIN_CONTENT_MAX_WIDTH,
```

Comment:
@mohandast52 lemme know if this is wrong, I don't see any issues, but maybe i'm missing something.